### ISSUE TITLE
Perform several network requests concurrently

### ISSUE THREAD

**Issue by ahhsfj1991** [LukasLechnerDev/Kotlin-Coroutine-Use-Cases-on-Android#3]
```
coroutineScope {
                    val oreoFeaturesDeferred = async { mockApi.getAndroidVersionFeatures(27) }
                    val pieFeaturesDeferred = async { mockApi.getAndroidVersionFeatures(28) }
                    val android10FeaturesDeferred = async { mockApi.getAndroidVersionFeatures(29) }

                    val oreoFeatures = oreoFeaturesDeferred.await()
                    val pieFeatures = pieFeaturesDeferred.await()
                    val android10Features = android10FeaturesDeferred.await()

                    val versionFeatures = listOf(oreoFeatures, pieFeatures, android10Features)

                    // other alternative: (but slightly different behavior when a deferred fails, see docs)
                    // val versionFeatures = awaitAll(oreoFeaturesDeferred, pieFeaturesDeferred, android10FeaturesDeferred)

                    uiState.value = UiState.Success(versionFeatures)
                }
```
i have some question about this code, deferred's await() function will suspend this coroutines, so this demo code can't performed in parallel

**Comment by LukasLechnerDev** [comment]
Hi! Yes `.await()` will suspend the parent coroutine, but the child coroutines started with `async{}` will keep running concurrently.

**Comment by ahhsfj1991** [comment]
Yes, the first `async()` executed will suspend the parent coroutine, then the next one will have no chance execute until the parent coroutine resumed. So it won't keep running concurrently.
